picky-asn1-der/src/application_tag.rs

```diff
@@ -0,0 +1,129 @@
+use crate::misc::Length;
+use crate::Asn1RawDer;
+use picky_asn1::tag::{Tag, TagPeeker};
+use serde::de::{Error, SeqAccess};
+use serde::{de, ser};
+use std::fmt;
+use std::fmt::Debug;
+
+#[derive(Debug, PartialEq)]
+pub struct ApplicationTag<V: Debug + PartialEq, const T: u8>(V);
+
+impl<V: Debug + PartialEq, const T: u8> ApplicationTag<V, T> {
+    pub fn from(value: V) -> Self {
+        Self(value)
+    }
+}
+
+impl<'de, V: de::Deserialize<'de> + Debug + PartialEq, const T: u8> de::Deserialize<'de> for ApplicationTag<V, T> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        struct Visitor<E>(Option<E>, u8);
```

Comment:
Why `Option<E>`? Also the tag could stay as a const generic

Reply:
This is a workaround. It allows me to specify `type Value = E;`(see below) otherwise I should specify `type Value = V` that gives the `error: generic parameter from outer function` compile error. Maybe, you have a better solution. :slightly_smiling_face:

Reply:
`PhantomData` is an idiomatic workaround 😄